addon/as-task.ts

```diff
@@ -0,0 +1,18 @@
+type GeneratorFn<Args extends any[] = any[], R = any> = (
+  ...args: Args
+) => IterableIterator<R>;
+
+interface Task<Args extends any[], R> {
+  perform(...args: Args): Promise<R>;
```

Comment:
This not only returns a `Promise`, but a `TaskInstance` that is compatible with the `Promise` interface and adds some additional props, like `isRunning`.

```suggestion
  perform(...args: Args): Promise<R> & TaskInstance<...>;
```